plot_continuous_monitor.py

```diff
@@ -226,9 +153,171 @@ def get_util_for_disk(disk_utils, disk):
+  """
+  Creates modified copies of gnuplot files, and associated pdf graphs for several
+  attributes from the continuous monitor.
+  """
+  # Get the location of the monotasks-scripts repository by getting the directory containing the
+  # file that is currently being executed.
+  scripts_dir = os.path.dirname(inspect.stack()[0][1])
+  attributes = ['utilization', 'disk_utilization', 'monotasks', 'memory']
+
+  for attribute in attributes:
+    plot_gnuplot_attribute(file_prefix, attribute, scripts_dir)
+
+  if open_graphs:
+    for attribute in attributes:
+      subprocess.check_call('open {0}_{1}.pdf'.format(file_prefix, attribute), shell=True)
+
+def plot_gnuplot_attribute(file_prefix, attribute, scripts_dir):
+  """ Create a gnuplot file and associated pdf for a some attribute (like disk_utilization) """
+  data_filename = '{0}_utilization'.format(file_prefix)
+  plot_filename = '{0}_{1}.gp'.format(file_prefix, attribute)
+  pdf_filename = '{0}_{1}.pdf'.format(file_prefix, attribute)
+  plot_file = open(plot_filename, 'w')
+
+  for line in open(os.path.join(scripts_dir, 'gnuplot_files/plot_{0}_base.gp'.format(attribute)), 'r'):
+    new_line = line.replace('__OUT_FILENAME__', pdf_filename).replace('__NAME__', data_filename)
+    plot_file.write(new_line)
+  plot_file.close()
+
+  subprocess.check_call('gnuplot {0}'.format(plot_filename), shell=True)
+
+def plot_matplotlib(cm_data, file_prefix, open_graphs):
+  disk_utilization_params = ['xvdf utilization',
+                             'xvdb utilization',
+                             'xvdf write throughput',
+                             'xvdf read throughput',
+                             'xvdb write throughput',
+                             'xvdb read throughput']
+  memory_params = ['free heap memory',
+                   'free off heap memory']
+  monotasks_params = ['local running macrotasks',
+                      'macrotasks in network',
+                      'macrotasks in compute',
+                      'macrotasks in disk',
+                      'running monotasks',
+                      'gc fraction',
+                      'outstanding network bytes']
+  utilization_params = ['cpu utilization',
+                        'xvdf utilization',
+                        'xvdb utilization',
+                        'cpu system',
+                        'gc fraction']
+  xvdf_params = ['xvdf running disk monotasks',
+                 'xvdf write throughput',
+                 'xvdf read throughput',
+                 'xvdf utilization']
+  xvdb_params = ['xvdb running disk monotasks',
+                 'xvdb write throughput',
+                 'xvdb read throughput',
+                 'xvdb utilization']
+
+  def plot_params(params_to_plot, title):
+    """
+    Creates a matplotlib graph using continuous monitor data.
+    Time is the x axis and data corresponding to each parameter is used to
+    generate a new line on the line graph.
```

Comment:
How about "Creates a matplotlib graph that plots a line for each parameter in params_to_plot as a function of time."